### PR TITLE
Update for PHP 7.1

### PR DESCRIPTION
**Pull request by shadowhand**
Adds strict typing and makes a couple of modifications for better type checking:

- `Pipeline::__construct` argument order changed to use variadic.
- `Processor::process` argument order changed to use variadic.

Also upgrades to latest phpspec and switches to a supported code coverage extension.

### CODE REVIEW

File: src/PipelineBuilderInterface.php [hunk 1]
```diff
@@ -1,23 +1,17 @@
 <?php
+declare(strict_types=1);
```
Comment (nyamsprod) [inline src/PipelineBuilderInterface.php:2]: If I remember correctly interfaces do not need the `declare` declaration since no code is executed in these files anyway

File: spec/PipelineBuilderSpec.php [hunk 2]
```diff
@@ -14,11 +14,6 @@ function it_is_initializable()
         $this->shouldHaveType(PipelineBuilder::class);
     }
 
-    function it_should_build_a_pipeline()
-    {
-        $this->build()->shouldHaveType(PipelineInterface::class);
-    }
```
Comment (frankdejonge) [inline spec/PipelineBuilderSpec.php:20]: Why is this removed?

File: src/PipelineBuilderInterface.php (hunk 1 above)
Comment (shadowhand) [reply src/PipelineBuilderInterface.php:2]: That *may* be the case, but I think it is better to have it everywhere for consistency.

File: spec/PipelineBuilderSpec.php (hunk 2 above)
Comment (shadowhand) [reply spec/PipelineBuilderSpec.php:20]: No need for it, the strict type already ensures return type without a test.

Comment (frankdejonge) [reply spec/PipelineBuilderSpec.php:20]: Fair enough 👍

**Comment by shadowhand** [comment]
Tests are now passing.

File: src/PipelineBuilderInterface.php (hunk 1 above)
Comment (frankdejonge) [reply src/PipelineBuilderInterface.php:2]: No value is lost or added with the removal or inclusion of this line of code 🤷‍♂️ it's all good for me

File: src/PipelineBuilderInterface.php [hunk 3]
```diff
@@ -1,23 +1,17 @@
...
 {
     /**
      * Add an stage.
-     *
-     * @param callable $stage
-     *
-     * @return $this
      */
-    public function add(callable $stage);
+    public function add(callable $stage): self;
```
Comment (frankdejonge) [inline src/PipelineBuilderInterface.php]: There's currently no way to express current instance returns in return types. This will effectively resolve as `PipelineBuilderInterface` which is clearer as a return type IMO. But the docblocks help editors to better reflect, since the return type can't.

Comment (shadowhand) [reply src/PipelineBuilderInterface.php]: Updated.
